crates/runtime/src/http/v1/workers.rs

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+use std::sync::Arc;
+
+use app::App;
+use axum::{
+    Extension,
+    extract::Query,
+    http::status,
+    response::{IntoResponse, Json, Response},
+};
+use csv::Writer;
+use serde::{Deserialize, Serialize};
+use tokio::sync::RwLock;
+
+use crate::Runtime;
+
+use super::Format;
+
+#[derive(Debug, Deserialize)]
+#[cfg_attr(feature = "openapi", derive(utoipa::IntoParams))]
+pub struct WorkersQueryParams {
+    /// The format of the response (e.g., `json` or `csv`).
+    #[serde(default)]
+    pub format: Format,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[cfg_attr(feature = "openapi", derive(utoipa::ToSchema))]
+#[serde(rename_all = "snake_case")]
+pub(crate) struct WorkerResponse {
+    object: String,
+    data: Vec<Worker>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[cfg_attr(feature = "openapi", derive(utoipa::ToSchema))]
+#[serde(rename_all = "snake_case")]
+pub(crate) struct Worker {
+    name: String,
+    description: Option<String>,
+
+    /// Use generic to avoid using `utoipa::ToSchema` in `spicepod` crate.
+    models: Vec<serde_json::Value>,
+}
+
+impl From<&spicepod::component::worker::Worker> for Worker {
+    fn from(value: &spicepod::component::worker::Worker) -> Self {
+        let spicepod::component::worker::Worker {
+            name,
+            description,
+            models,
+            ..
+        } = value;
+
+        Worker {
+            name: name.clone(),
+            description: description.clone(),
+            models: models
+                .iter()
+                .filter_map(|m| serde_json::to_value(m).ok())
+                .collect(), // This cannot fail as it has already been serialized.
```

Comment:
I'm not sure what this comment means

Reply:
Sorry on wrong line, i've improved comment too.